### ISSUE TITLE
testing - rackup folder - sleep

### ISSUE THREAD

**Issue by MSP-Greg** [puma/puma#1902]
There are several files in the test/rackup folder, and a few make use of `sleep`, with varying times.

The same thing may be done when 'app' lambdas are declared in a test method.  I've worked locally and in my fork with a file similar to:

```ruby
run lambda { |env|
  dly = (env['REQUEST_PATH'][/\/(d+)sec/,1] || '0').to_i
  sleep dly
  [200, {}, ["Delayed #{dly}"]]
}
```
as a replacement, and also used a constant lambda defined in `helper.rb'.

This came about when thinking of tests for external KILL/TERM, graceful shutdown, phased-restarts, etc.  A way to clearly create requests with specific time delays seems needed.

The above needs a GET of '\/<x\>sec', where \<x\> is an integer, and replies with 'Delayed \<x>\'.

Hence, two questions:

1. Stupid idea or otherwise?
2. If a 'good' idea', what to name the file, the constant, the GET and the reply?  Kind of sensitive to ESL issues, as contributors are worldwide...

**Comment by nateberkopec** [comment]
Makes sense. 

I think we could add it a new "helpers/apps.rb" file that's required by test/helper. Namespace something like `ExampleRackApps::SLEEP`. I think you should just use the world `sleep` as it describes what the app actually does, which is call sleep.

**Comment by dentarg** [comment]
Can this issue be closed now that #1903 was merged?

**Comment by MSP-Greg** [comment]
Ok with me...
